src/board.cpp

```diff
@@ -875,6 +882,9 @@ void initBitmaphelper()
     }
 }
 
+} // namespace rubichess
+
+
```

Comment:
In a .cpp file, a namespace could be used to pack all global variants and functions but not class functions (because classes are declared in a namespace from the header file). In board.cpp, global variants and functions are separated by class functions, thus a few namespaces have been added since I have managed to minimise changing the code.

It is better if we group those global variants/functions in one place. However, if I did that, you may see so many changes and get confused.

You may group them now or wait for other steps, say, make sure all work.

Reply:
If you could and if it doesn't affect the speed/strength, reduce the number of global variants, and functions, by putting them into some classes.
Ideally, especially for libraries/sharing purposes, all code of an engine should be in one class thus we can create and delete it easily and completely from memory:

```
RubiChess* engine = RubiChess();
engine->loop();
delete engine;
```

Reply:
IMHO, for modern computers, a little decrease in speed doesn't matter (it may lose under 1 Elo). It is especially true when your chess engine has used NNUE.